nullaway/src/main/java/com/uber/nullaway/handlers/InferredJARModelsHandler.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.handlers;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.google.errorprone.VisitorState;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.tools.javac.code.Symbol;
+import com.uber.nullaway.NullAway;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/** This handler loads inferred nullability model from stubs for methods in unannotated packages. */
+public class InferredJARModelsHandler extends BaseNoOpHandler {
+
+  public static final int VERSION_0_FILE_MAGIC_NUMBER = 691458791;
+  private static final boolean VERBOSE = false;
+  private static final boolean DEBUG = false;
+
+  private static final String DEFAULT_ASTUBX_LOCATION = "META-INF/nullaway/jarinfer.astubx";
+  private static Map<String, Map<String, Map<Integer, Set<String>>>> argAnnotCache;
+
+  public InferredJARModelsHandler() {
+    super();
+    argAnnotCache = new LinkedHashMap<>();
+  }
+
+  @Override
+  public ImmutableSet<Integer> onUnannotatedInvocationGetNonNullPositions(
+      NullAway analysis,
+      VisitorState state,
+      Symbol.MethodSymbol methodSymbol,
+      List<? extends ExpressionTree> actualParams,
+      ImmutableSet<Integer> nonNullPositions) {
+
+    String className = methodSymbol.enclClass().getQualifiedName().toString();
+    if (DEBUG) {
+      System.out.println("[JI DEBUG] class: " + className);
```

Comment:
So, ideally, we want either these all gone or some sort of global config option that turns these on or off. Not sure if EP provides that, or NullAway itself for that matter. In any case, maybe this is fine for debugging for now and just have them all removed by the compiler on a normal build.

@msridhar ? (These are kinda useful for debugging the unit tests, and `DEBUG`/`VERBOSE` are `false` constants as committed, so they won't end up in the build logs, so this is not urgent for this PR. But I was wondering if you had any suggestions)

@subarnob : In any case, this is probably not a blocker for this PR.